testng-core/src/test/java/test/dependent/DependentTest.java

```diff
@@ -227,4 +206,84 @@ public void simpleCyclingDependencyShouldWorkWitParallelism(
                 + Arrays.deepToString(runMethods));
     assertThat(listener.getSucceedMethodNames()).containsExactlyInAnyOrder(runMethods[0]);
   }
+
+  @Test(description = "GITHUB-2658")
+  public void testMethodDependencyAmidstInheritance() {
+    TestNG testng = new TestNG();
```

Comment:
Why not using SimpleBaseTest.create here?

Reply:
That was a copy paste error. Fixed.